third_party/terraform/resources/resource_cloudfunctions_function.go

```diff
@@ -550,9 +554,28 @@ func flattenEventTrigger(eventTrigger *cloudfunctions.EventTrigger) []map[string
 		return result
 	}
 
+	resource := ""
+	switch {
+	case strings.HasPrefix(eventTrigger.EventType, "google.storage.object."):
+		resource = GetResourceNameFromSelfLink(eventTrigger.Resource)
+	case strings.HasPrefix(eventTrigger.EventType, "google.pubsub.topic."):
+		resource = GetResourceNameFromSelfLink(eventTrigger.Resource)
+		// Legacy style triggers
+	case strings.HasPrefix(eventTrigger.EventType, "providers/cloud.storage/eventTypes/"):
+		resource = GetResourceNameFromSelfLink(eventTrigger.Resource)
+	case strings.HasPrefix(eventTrigger.EventType, "providers/cloud.pubsub/eventTypes/"):
+		resource = GetResourceNameFromSelfLink(eventTrigger.Resource)
+	case strings.HasPrefix(eventTrigger.EventType, "providers/cloud.firestore/eventTypes/"):
+		// Simply taking the substring after the last "/" is not sufficient for firestore as resources may have slashes.
+		// For the eventTrigger.Resource "projects/my-project/databases/(default)/documents/messages/{messageId}" we extract
+		// the resource "messages/{messageId}" by taking the everything after the 5th "/"
+		parts := strings.SplitN(eventTrigger.Resource, "/", 6)
```

Comment:
This feels a bit brittle. You could change to a regex and at least that would document the expected pattern and fail to find anything rather than inadvertently pulling in an unexpected string. Also consider throwing an error for missing or unexpected input.

Reply:
Ehh I don't think this is worth using a regex for. I don't think the risk of relative uris changing in shape is a big problem, regex code is more complicated, and we've already documented with a comment. If you feel strongly feel free to PR in a change to use one though.